rllib/policy/tf_policy.py

```diff
@@ -162,7 +163,21 @@ def __init__(self,
         else:
             logger.info("TFPolicy (worker={}) running on {} GPU(s).".format(
                 worker_idx if worker_idx > 0 else "local", num_gpus))
-            gpu_ids = ray.get_gpu_ids()
+
+            # We are a remote worker (WORKER_MODE=1):
+            # GPUs should be assigned to us by ray.
+            if ray.worker._mode() == ray.worker.WORKER_MODE:
+                gpu_ids = ray.get_gpu_ids()
+
+            if not gpu_ids:
+                raise ValueError(
+                    "TFPolicy was not able to find any GPU IDs, even "
+                    "though torch.cuda.is_available()=True!")
+            elif len(gpu_ids) < num_gpus:
+                raise ValueError(
+                    "TFPolicy was not able to find enough GPU IDs! Found "
+                    f"{gpu_ids}, but num_gpus={num_gpus}.")
```

Comment:
I think we should use `if len(self.devices) > 0` bellow. This condition fails on `num_gpus=0.5`. `for i, _ in enumerate(...) if i < num_gpus` can handle fractional GPUs.

Reply:
Not sure this would be necessary:

E.g.:
if num_gpus=0.5 and gpu_ids=["/physical_device:gpu:0"]

then this tf check would pass, no (and the error would not be raised)?

Also:
```
self.devices = [f"/gpu:{i}" for i, _ in enumerate(gpu_ids) if i < num_gpus]
```
would still generate a device list with exactly 1 gpu in it despite num_gpu being 0.5.